src/api/app/helpers/webui/notification_helper.rb

```diff
@@ -16,6 +18,78 @@ def filter_notification_link(link_text, amount, filter_item)
     end
   end
 
+  def request_badge_color(state)
+    case state
+    when :review, :new
+      'secondary'
+    when :declined, :revoke
+      'danger'
+    when :superseded
+      'warning'
+    when :accepted
+      'success'
+    else
+      'dark'
+    end
+  end
+
+  def notification_link(notification)
+    link_text = case notification.event_type
+                when 'Event::RequestStatechange', 'Event::RequestCreate', 'Event::ReviewWanted'
+                  "Request ##{notification.notifiable.number}"
+                when 'Event::CommentForRequest'
+                  "Request ##{notification.notifiable.commentable.number}"
+                when 'Event::CommentForProject'
+                  notification.notifiable.commentable.name
+                when 'Event::CommentForPackage'
+                  commentable = notification.notifiable.commentable
+                  "#{commentable.project.name} / #{commentable.name}"
+                end
+
+    link_to(link_text, notification.link_to_notification_target, class: 'mx-1 text-word-break-all')
+  end
+
+  def read_unread_link(notification)
+    title, icon = notification.unread? ? ['Mark as "Read"', 'fa-check'] : ['Mark as "Unread"', 'fa-undo']
+
+    link_to(my_notification_path(id: notification),
+            method: :put, class: 'btn btn-sm btn-outline-success px-3', title: title, remote: true) do
+      concat(tag.i(class: "fas #{icon}"))
+    end
+  end
+
+  # A limited number of avatars are going to be displayed
+  def request_tags(request)
```

Comment:
This really sounds like a scope or method in `BsRequest` that tells you the involved people instead. 

So instead of building an array of image tags in a helper, you would ask the request for the involved people and create the HTML in a loop.

I'm pretty sure the `BsRequest` method will be useful for other things too.

Reply:
That was my initial plan and at some point I thought it wasn't a good idea. I'll give it a second thought.

Reply:
This code has been refactored. Maybe is not exactly what you requested, but the code has been simplified a lot.
We can refactor this in another round.